### ISSUE TITLE
Unable to build with fresh new install of Fedora 31

### ISSUE THREAD

**Issue by Dimitri2g** [ValveSoftware/steam-runtime#259]
I did a git clone and have been following the instructions and it seems I am failing to build.
Instructions:
[Instructions](https://github.com/ValveSoftware/Proton/wiki/Build-instructions-(Fedora))

I ran:
`./steam-runtime/setup_docker.sh --beta amd64 --extra-bootstrap=./proton/steamrt-bootstrap.sh steam-proton-dev`

and at the end I got:

```---> 27c0d708c8e6
Step 8/20 : RUN 	set -xe; 	if [ -n "${proxy}" ]; then 		echo "configure proxy"; 		echo "export http_proxy=${proxy}" >> /etc/bash.bashrc; 		echo "export https_proxy=${proxy}" >> /etc/bash.bashrc; 		echo "export ftp_proxy=${proxy}" >> /etc/bash.bashrc; 		echo "Acquire::http::proxy \"${proxy}\";" > /etc/apt/apt.conf.d/01proxy; 		echo "Acquire::https::proxy \"${proxy}\";" >> /etc/apt/apt.conf.d/01proxy; 	fi; 	if [ -n "${no_proxy}" ]; then 		echo "export no_proxy=${no_proxy}" >> /etc/bash.bashrc; 		oldIFS="$IFS"; 		IFS=","; 		for nope in ${no_proxy}; do 			case "$nope" in 				(.*) 					;; 				(*) 					echo "Acquire::http::proxy::${nope} \"DIRECT\";" >> /etc/apt/apt.conf.d/01proxy; 		echo "Acquire::https::proxy::${nope} \"DIRECT\";" >> /etc/apt/apt.conf.d/01proxy; 					;; 			esac; 		done; 		IFS="$oldIFS"; 	fi
 ---> Running in e2cfab95578d
OCI runtime create failed: this version of runc doesn't work on cgroups v2: unknown

real	0m28.535s
user	0m0.645s
sys	0m0.481s
Script done, file is /tmp/steam-runtime-setup-chroot-Vm8.log
[steam-runtime-setup-chroot-Vm8.log](https://github.com/ValveSoftware/Proton/files/4536111/steam-runtime-setup-chroot-Vm8.log)
```

Full log attached
[steam-runtime-setup-chroot-Vm8.log](https://github.com/ValveSoftware/Proton/files/4536117/steam-runtime-setup-chroot-Vm8.log)

**Comment by smcv** [comment]
`setup_docker.sh` is deprecated. For the recommended approach, please see: https://github.com/ValveSoftware/steam-runtime#using-docker

However, that won't necessarily solve your problem.

> OCI runtime create failed: this version of runc doesn't work on cgroups v2: unknown

I would recommend trying a very simple Docker example, for example pulling and running https://hub.docker.com/_/hello-world. If that doesn't work either, report the problem to your distribution (Fedora).

If Fedora is actually using Podman instead of Docker, then running it unprivileged (without `sudo`) might be better?

**Comment by dreamer** [comment]
Yeah, I haven't updated those wiki instructions in a long time… and I don't have time to get over them and fix it myself at the moment.

As for the problem at hand: it's most likely caused by Docker still not understanding cgroups v2, and this API was turned on by default in Fedora 31. There are two ways to fix it:

- Use `podman` instead of Docker. It's a Docker fork with issues like these fixes, it's included in Fedora repositories and should work as a drop-in replacement for docker.
- Force-disable cgroupsv2

[Here's an article that seems to cover both options in detail](https://www.linuxuprising.com/2019/11/how-to-install-and-use-docker-on-fedora.html)

**Comment by Dimitri2g** [comment]
@smcv  and @dreamer Thank you both for responding.

I will try to use podman and disable cgroupsv2

**Comment by dreamer** [comment]
@Dimitri2g With podman you won't need to disable cgroupsv2.
